cognite/neat/_rules/transformers/_converters.py

```diff
@@ -220,6 +224,140 @@ def _transform(self, rules: VerifiedRules) -> VerifiedRules:
         raise ValueError(f"Unsupported conversion from {type(rules)} to {self._out_cls}")
 
 
+_T_Entity = TypeVar("_T_Entity", bound=ClassEntity | ViewEntity)
+
+
+class ToExtension(RulesTransformer[DMSRules, DMSRules]):
+    def __init__(
+        self, new_model_id: DataModelIdentifier, prefix: str | None = None, mode: Literal["composition"] = "composition"
+    ):
+        self.new_model_id = DataModelId.load(new_model_id)
+        self.prefix = prefix
+        self.mode = mode
+
+    def transform(self, rules: DMSRules | OutRules[DMSRules]) -> JustRules[DMSRules]:
+        # Copy to ensure immutability
+        verified = self._to_rules(rules)
+        if self.prefix is None and verified.metadata.as_data_model_id() in COGNITE_MODELS:
+            raise NeatValueError(f"Prefix is required when extending {verified.metadata.as_data_model_id()}")
+        source_id = verified.metadata.as_data_model_id()
+
+        dump = verified.dump()
+        dump["metadata"]["schema_"] = SchemaCompleteness.partial.value
+        dump["metadata"]["space"] = self.new_model_id.space
+        dump["metadata"]["external_id"] = self.new_model_id.external_id
+        if self.new_model_id.version is not None:
+            dump["metadata"]["version"] = self.new_model_id.version
+        new_model = DMSRules.model_validate(DMSInputRules.load(dump).dump())
+
+        for prop in new_model.properties:
+            if prop.container and prop.container.space == self.new_model_id.space:
```

Comment:
I guess you need this since when you are dumping new data model id, container which do not have explicit statement of space will be automatically set to belong to new space, so you are reverting them back to the original config.

If so, just a comment above this for-loop would make code more readable